src/skeleton/read.rs

```diff
@@ -35,87 +35,63 @@ pub(super) fn config<P: AsRef<Path>>(base_path: &P) -> Result<Option<String>, an
     }
 }
 
-fn vendored_directory(config_contents: Option<&str>) -> Option<String> {
-    let contents = config_contents.and_then(|contents| contents.parse::<toml::Value>().ok())?;
-    let source = contents.get("source")?;
-    let crates_io = source.get("crates-io")?;
-    let vendored_field_suffix = crates_io
-        .get("replace-with")
-        .and_then(|value| value.as_str())?;
-    let vendored_sources = source.get(vendored_field_suffix)?;
-    Some(vendored_sources.get("directory")?.as_str()?.to_owned())
-}
-
 pub(super) fn manifests<P: AsRef<Path>>(
     base_path: &P,
-    config_contents: Option<&str>,
+    metadata: Metadata,
 ) -> Result<Vec<ParsedManifest>, anyhow::Error> {
-    let vendored_path = vendored_directory(config_contents);
-    let builder = if let Some(path) = vendored_path {
-        let exclude_vendored_sources = format!("!{}", path);
-        GlobWalkerBuilder::from_patterns(base_path, &["/**/Cargo.toml", &exclude_vendored_sources])
-    } else {
-        GlobWalkerBuilder::new(base_path, "/**/Cargo.toml")
-    };
-    let walker = builder
-        .build()
-        .context("Failed to scan the files in the current directory.")?;
+    let mut metadata_paths = metadata
+        .workspace_packages()
+        .into_iter()
+        .map(|package| package.manifest_path.clone().into_std_path_buf())
+        .collect::<HashSet<_>>();
+    metadata_paths.insert(base_path.as_ref().join("Cargo.toml"));
+    let mut manifest_paths: Vec<PathBuf> = metadata_paths.into_iter().collect();
+    manifest_paths.sort();
```

Comment:
We can avoid this "collect" + "collect" + "sort" by using a `BTreeSet` directly.